lalrpop/src/rust/mod.rs

```diff
@@ -106,14 +107,21 @@ impl<W:Write> RustWrite<W> {
 
     pub fn write_pub_fn_header(&mut self,
                                grammar: &Grammar,
+                               visibility: &Visibility,
                                name: String,
                                type_parameters: Vec<String>,
                                parameters: Vec<String>,
                                return_type: String,
                                where_clauses: Vec<String>)
                                -> io::Result<()>
     {
-        self.write_fn_header_helper(grammar, "pub ", name, type_parameters,
+        let path = match *visibility {
+            Visibility::Pub(ref path) => path.as_ref().map(|p| format!("pub({}) ", p)),
+            Visibility::Priv => panic!(),
+        };
+        self.write_fn_header_helper(grammar,
+                                    &path.as_ref().map(|p| p.as_str()).unwrap_or("pub "),
```

Comment:
why don't we make `write_fn_header_helper` take the `Visibility` instead of a string?